cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/MonitorHandler.java

```diff
@@ -169,6 +187,7 @@ public void getBootStatus(HttpRequest request, HttpResponder responder) {
   @Path("/system/services")
   @GET
   public void getServiceSpec(HttpRequest request, HttpResponder responder) {
+    contextAccessEnforcer.enforceOnParent(EntityType.SYSTEM_SERVICE, new InstanceId(""), StandardPermission.LIST);
```

Comment:
Nit: we should make this `new InstanceId("")` a factory method. I hate seeing this empty string constant all over the code

Reply:
Agreed, I went ahead and made the instance ID referring to the current instance a public static final constant in the InstanceId class similar to `NamespaceId.SYSTEM`.

Reply:
Hurray!